lightning/src/ln/channelmanager.rs

```diff
@@ -5828,10 +5834,50 @@ where
 					None => continue 'outer_loop,
 				}
 
+				let mut inter_trampoline_path = None;
+
 				// Now process the HTLC on the outgoing channel if it's a forward.
 				if let Some(next_packet_details) = next_packet_details_opt.as_ref() {
+					let first_hop_scid_override = match (&next_packet_details.outgoing_connector, &next_hop) {
+						(HopConnector::Trampoline(next_node_id), onion_utils::Hop::TrampolineForward { ref outer_hop_data, .. } | onion_utils::Hop::TrampolineBlindedForward { ref outer_hop_data, .. }) => {
+							let incoming_amount = outer_hop_data.amt_to_forward;
+							let incoming_cltv = outer_hop_data.outgoing_cltv_value;
+							let outgoing_amount = next_packet_details.outgoing_amt_msat;
+							let outgoing_cltv = next_packet_details.outgoing_cltv_value;
+
+							let routes = self.router.find_route(
```

Comment:
Is there any risk in this potentially long running operation blocking important processes?

Reply:
It is running in the background processor I think, but it could still be blocking the processing of other events? Or is execution really suspended here while other event handling proceeds?